doc/6/getting-started/react/with-redux/index.md

```diff
@@ -0,0 +1,171 @@
+
+<<< ./snippets/App.js.snippet:6[js]
+
+After that, create that function with the connection to Kuzzle:
+
+<<< ./snippets/App.js.snippet:2[js]
+
+Then we will [etablish the connection](/sdk/js/6/core-classes/kuzzle/connect/) to kuzzle and create, if they don't [exists](sdk/js/6/controllers/index/exists/), the [index](sdk/js/6/controllers/index/create/) and [collection](sdk/js/6/controllers/collection/create/) of our chat. 
+
+Add the following lines to the `_initialize` function:
+
+<<< ./snippets/App.js.snippet:3[js]
+
+## Display the messages
+
+We'll need some properties and functions to manage our messages.
+
+We have to create our Redux store architecture (more details on [Redux documentation](https://redux.js.org/introduction/getting-started)), like this:
+
+
+```bash
+src
+└── state
+    ├── actions.js
+    ├── reducers.js
+    └── sagas.js
+```
+
+
+Add the following actions to the _src/state/actions.js_ file:
+
+<<< ./snippets/actions.js.snippet[js]
+
+Then we'll edit the _src/state/reducers.js_ file:
+
+Add the `initialState`:
+
+<<< ./snippets/reducers.js.snippet:1[js]
+
+And the `reducersMap` with our `SET_MESSAGE` action:
+
+<<< ./snippets/reducers.js.snippet:2[js]
+
+Finally, export it: 
+
+<<< ./snippets/reducers.js.snippet:3[js]
+
+The entire file should look like this:
+
+<<< ./snippets/reducers.js.snippet[js]
+
+Now that our store is ready, we'll fetch the existing messages in Kuzzle and add them to our store.
+Add the following lines to the `_initialize()` function of the `app` class in the _src/App.js_ file:
+
+<<< ./snippets/App.js.snippet:7[js]
+
+Then, add the following constants in the `render()` function of the `app class:
+
+<<< ./snippets/App.js.snippet:11[js]
+
+And the loop in the return of the `render()` function to display the messages stored:
+
+<<< ./snippets/App.js.snippet:13[html]
+
+
+We can now display the messages stored in Kuzzle. Cool but we need to create some right ?
+
+## Send messages
+
+We need to write a simple method that will [create](/sdk/js/6/controllers/document/create/) a new message document in Kuzzle.
+Add the following function in your `app` class in the_src/App.js_ file:
+
+<<< ./snippets/App.js.snippet:10[js]
+
+Then, we need to create the `sendMessage()` Redux action we just called.
+_src/state/sagas.js_ contains a generator function where we will put our sagas function. (more details on [Redux-saga documentation](https://redux-saga.js.org/)):
+
+Let's add it in the _src/state/sagas.js_ file:
+
+<<< ./snippets/sagas.js.snippet[js]
+
+As you can see we don't push the new message in our state on message creation.
+Actually, We'll subscribe to the collection that contains our messages.
+So let's create our `_subscribeToNewMessages()` function in the `app` class in _src/App.js_ file. It will call the realtime controller of Kuzzle to allow us to [receive notifications](/sdk/js/6/controllers/realtime/subscribe/) on message creation:
```

Comment:
```suggestion
So let's create our `_subscribeToNewMessages()` function in the `app` class in _src/App.js_ file. It will call Kuzzle's realtime controller to allow us to [receive notifications](/sdk/js/6/controllers/realtime/subscribe/) on message creations:
```